homeworks/Makc.Sas_MakcSas/homework_6/homework_6_1.js

```diff
@@ -0,0 +1,21 @@
+const user = {};
+Object.defineProperties(user, {
+    fullName: {
+        set: (name) => {
+            this.fullName = name.replace(/([[A-Za-z])\S/g, s => s.toLowerCase()).replace(/(^|\s|[-])\S/g, s => s.toUpperCase());
```

Comment:
arrow function will effect global scope here, also your RegExp could be a bit lighter here, you could check for first letters of words, something like `\b\w`